### PR TITLE
[lmi] Upgrade lmi dockerfile for 0.29.0 release

### PR DESCRIPTION
**Pull request by maaquib**
## Description ##

CI test: https://github.com/deepjavalibrary/djl-serving/actions/runs/9916087551

- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: serving/docker/lmi.Dockerfile [hunk 1]
```diff
@@ -19,22 +19,22 @@ ARG torch_version=2.3.0
...
-ARG datasets_version=2.19.1
+ARG datasets_version=2.20.0
 ARG autoawq_version=0.2.5
+ARG tokenizers_version=0.19.1
 # LMI-Dist Deps
-ARG vllm_wheel="https://publish.djl.ai/vllm/cu124-pt230/vllm-0.4.2%2Bcu124-cp310-cp310-linux_x86_64.whl"
-ARG flash_attn_2_wheel="https://publish.djl.ai/flash_attn/cu124-pt230/flash_attn-2.5.8-cp310-cp310-linux_x86_64.whl"
+ARG vllm_wheel="https://github.com/vllm-project/vllm/releases/download/v0.5.0.post1/vllm-0.5.0.post1-cp310-cp310-manylinux1_x86_64.whl"
+ARG flash_attn_2_wheel="https://github.com/vllm-project/flash-attention/releases/download/v2.5.9/vllm_flash_attn-2.5.9-cp310-cp310-manylinux1_x86_64.whl"
+ARG flash_infer_wheel="https://github.com/flashinfer-ai/flashinfer/releases/download/v0.0.8/flashinfer-0.0.8+cu121torch2.3-cp310-cp310-linux_x86_64.whl"
```
Comment (lanking520) [inline serving/docker/lmi.Dockerfile:37]: does this compiled to all SM archs? 80/86/89/90?

File: serving/docker/lmi.Dockerfile [hunk 2]
```diff
@@ -19,22 +19,22 @@ ARG torch_version=2.3.0
...
+ARG optimum_version=1.21.2
 ARG auto_gptq_version=0.7.1
-ARG datasets_version=2.19.1
+ARG datasets_version=2.20.0
 ARG autoawq_version=0.2.5
+ARG tokenizers_version=0.19.1
 # LMI-Dist Deps
-ARG vllm_wheel="https://publish.djl.ai/vllm/cu124-pt230/vllm-0.4.2%2Bcu124-cp310-cp310-linux_x86_64.whl"
-ARG flash_attn_2_wheel="https://publish.djl.ai/flash_attn/cu124-pt230/flash_attn-2.5.8-cp310-cp310-linux_x86_64.whl"
+ARG vllm_wheel="https://github.com/vllm-project/vllm/releases/download/v0.5.0.post1/vllm-0.5.0.post1-cp310-cp310-manylinux1_x86_64.whl"
```
Comment (lanking520) [inline serving/docker/lmi.Dockerfile:35]: why we still use 0.5.0.post1? Should we upgrade to 0.5.1?

Comment (sindhuvahinis) [reply serving/docker/lmi.Dockerfile:35]: Yes, multimodal support depends on 0.5.1 of vllm

File: serving/docker/lmi.Dockerfile (hunk 1 above)
Comment (maaquib) [reply serving/docker/lmi.Dockerfile:37]: https://github.com/flashinfer-ai/flashinfer/blob/v0.0.8/docs/installation.rst#prerequisites

File: serving/docker/lmi.Dockerfile (hunk 2 above)
Comment (maaquib) [reply serving/docker/lmi.Dockerfile:35]: This will be updated once we have lmi-dist support for 0.5.1

**Review by lanking520 — COMMENTED** [review]
We should also install autofp8 into our solution stack

File: tests/integration/llm/prepare.py [hunk 3]
```diff
@@ -447,17 +454,6 @@
         "option.tensor_parallel_degree": 4,
         "option.max_rolling_batch_size": 4
     },
-    "llama-7b-unmerged-lora": {
```
Comment (maaquib) [inline tests/integration/llm/prepare.py:450]: Removing duplicate

**Review by lanking520 — APPROVED** [review]
